File: python/n392_IsSequence.py
# https://leetcode-cn.com/problems/is-subsequence/

class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        i = j = 0
        s1 = len(s)
        s2 = len(t)
        if s2 < s1: return False
        if s1 == 0: return True
        while i < s1:
            if j >= s2: return False
            while j < s2 and t[j] != s[i]:
                j += 1
                if j >= s2:
                    return False
            i += 1
            j += 1
        return True 
        

if __name__ == "__main__":
    pass
            
        

File: cpp/n392_IsSequence.cpp
// https://leetcode-cn.com/problems/is-subsequence/
#include <iostream>
#include <string>
#include <map>
#include <vector>
using namespace std;

class Solution {
public:
    bool isSubsequence(string s, string t) {
        int i = 0, j = 0;
        int s1 = s.size();
        int s2 = t.size();
        if (s1 == 0) return true;
        while(i <= s1){
            while (t[j] != s[i]){
                j += 1;
                if (j > s2) return false;
            }
            i += 1;
            j += 1;
        }
        return true;
    }
};

int main(){
    return 0;
}